Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -3591,6 +3591,7 @@ - (NSArray *)supportedPasteboardTypesForCurrentSelection
 
     static NeverDestroyed plainTextTypes = [] {
```

Comment:
Is `plainTextTypes` still an accurate name?

Reply:
It's a _bit_ of a stretch, but I think it could still be appropriate in the sense that this is data that could be consumed by a paste event handler on the page. That said, the same applies to `UIPasteboardTypeListImage` as well but we still treat that as a rich type.

Maybe we should just change it to something like this?

```
- (NSArray *)supportedPasteboardTypesForCurrentSelection
{
    if (_page->editorState().selectionIsNone)
        return nil;

    static NeverDestroyed supportedTypes = [] {
        auto plainTextTypes = adoptNS([[NSMutableArray alloc] init]);
        [plainTextTypes addObject:@(WebCore::PasteboardCustomData::cocoaType().characters())];
ALLOW_DEPRECATED_DECLARATIONS_BEGIN
        [plainTextTypes addObject:(id)kUTTypeURL];
ALLOW_DEPRECATED_DECLARATIONS_END
        [plainTextTypes addObjectsFromArray:UIPasteboardTypeListString];
        return plainTextTypes;
    }();
    static NeverDestroyed supportedTypesWithImageAndWebArchive = [] {
        auto richTypes = adoptNS([[NSMutableArray alloc] init]);
        [richTypes addObject:WebCore::WebArchivePboardType];
        [richTypes addObjectsFromArray:UIPasteboardTypeListImage];
        [richTypes addObjectsFromArray:plainTextTypes.get().get()];
        return richTypes;
    }();

    if (_page->editorState().isContentRichlyEditable)
        return supportedTypesWithImageAndWebArchive.get().get();
    return supportedTypes.get().get();
}
```

(as an aside, we should probably allow `WebArchivePboardType` when pasting in plain text fields as well since we'd gracefully convert that to text content — but we can deal with that separately).